cdap-unit-test/src/test/java/io/cdap/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -1036,6 +1037,54 @@ public void testGetServiceURL() throws Exception {
     centralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
   }
 
+  @Category(SlowTests.class)
+  @Test
+  public void testGetServiceURLDiffNamespace() throws Exception {
+    ApplicationManager discoveringApplicationManager = deployApplication(AppUsingGetServiceURL.class);
+    ServiceManager discoveringServiceManager = discoveringApplicationManager
+      .getServiceManager(AppUsingGetServiceURL.FORWARDING).start();
+    discoveringServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    NamespaceId systemSpace = new NamespaceId("system");
+    ApplicationManager systemApplicationManager = deployApplication(systemSpace, AppWithServices.class);
+    ServiceManager systemCentralServiceManager =
+      systemApplicationManager.getServiceManager(AppWithServices.SERVICE_NAME).start();
+    systemCentralServiceManager.waitForRun(ProgramRunStatus.RUNNING, 10, TimeUnit.SECONDS);
+
+    String result = callServiceGet(discoveringServiceManager.getServiceURL(), "appWithServices");
+    String decodedResult = new Gson().fromJson(result, String.class);
+    Assert.assertEquals(AppWithServices.ANSWER, decodedResult);
+
+    discoveringServiceManager.stop();
+    systemCentralServiceManager.stop();
+
+    discoveringServiceManager.waitForStopped(10, TimeUnit.SECONDS);
+    systemCentralServiceManager.waitForStopped(10, TimeUnit.SECONDS);
+  }
+
+  @Category(SlowTests.class)
+  @Test
+  public void testGetServiceURLSameNamespace() throws Exception {
```

Comment:
this test looks like it's identical to the other one.

Reply:
Sort of. The difference is that, in this test, both apps are deployed in the same namespace. In the above test, the apps are deployed in different namespaces.

Reply:
oh I see there's only that one line that's different. I'm confused though, does the app hardcode 'system' as the namespace when discovering the service? How is this test passing?

Reply:
I don't think we really need this test. There are existing tests for within-namespace service discovery.

Reply:
Removed